--- src/CMakeLists.txt.orig	2014-02-21 14:59:17.503001754 +0100
+++ src/CMakeLists.txt	2014-02-21 15:04:08.565001709 +0100
@@ -244,10 +244,10 @@
 if(CGNS_BUILD_SHARED)
   # for windows, need to install both cgnsdll.dll and cgnsdll.lib
   if (WIN32 OR CYGWIN)
-    install(TARGETS cgns_shared ARCHIVE DESTINATION lib)
-    install(TARGETS cgns_shared RUNTIME DESTINATION lib)
+    install(TARGETS cgns_shared ARCHIVE DESTINATION ${LIB_SUFFIX})
+    install(TARGETS cgns_shared RUNTIME DESTINATION bin)
   else (WIN32 OR CYGWIN)
-    install(TARGETS cgns_shared LIBRARY DESTINATION lib)
+    install(TARGETS cgns_shared LIBRARY DESTINATION ${LIB_SUFFIX})
   endif (WIN32 OR CYGWIN)
 endif(CGNS_BUILD_SHARED)
 
