--- src/cgnstools/cgnsview/cgnsview.tcl.orig	2014-03-13 12:25:29.463483120 +0100
+++ src/cgnstools/cgnsview/cgnsview.tcl	2014-03-13 11:49:03.554483453 +0100
@@ -2196,6 +2196,10 @@
   }
 }
 
+proc makeAbsolute {pathname} {
+    file join [pwd] $pathname
+}
+
 proc file_load {{inpfile ""}} {
   global ProgData tcl_platform
   if {$inpfile == ""} {
@@ -2203,6 +2207,7 @@
       $ProgData(extensions)]
     if {$inpfile == ""} return
   }
+  set inpfile [makeAbsolute $inpfile]
   if [catch {CGNSfile $inpfile} type] {
     errormsg $type
     return
--- src/cgnstools/cgnsplot/cgnsplot.tcl.orig	2014-03-13 12:25:29.464483120 +0100
+++ src/cgnstools/cgnsplot/cgnsplot.tcl	2014-03-13 11:52:41.490483420 +0100
@@ -2150,6 +2150,10 @@
   {{All Files} {*}}
 }
 
+proc makeAbsolute {pathname} {
+    file join [pwd] $pathname
+}
+
 proc load_cgns {{filename ""}} {
   global ProgData FileList ModelData
   global Translate Scale
@@ -2159,6 +2163,9 @@
       $ProgData(cgnsfile) . $FileList]
   }
   if {$filename == ""} return
+
+  set filename [makeAbsolute $filename]
+
   cleanup_display
   display_message "reading base information..."
   . configure -cursor watch
