--- src/CMakeLists.txt.orig	2014-02-18 19:11:54.477038929 +0100
+++ src/CMakeLists.txt	2014-02-18 19:12:49.611038920 +0100
@@ -197,32 +197,33 @@
   # and link with the HDF5 libraries to build a dll
   if (WIN32 OR CYGWIN)
     add_definitions(-DBUILD_DLL)
-    if (ENABLE_HDF5 AND HDF5_LIBRARY)
-      target_link_libraries(cgns_shared ${HDF5_LIBRARY})
-      if(HDF5_NEED_ZLIB AND ZLIB_LIBRARY)
-        target_link_libraries(cgns_shared ${ZLIB_LIBRARY})
-      endif(HDF5_NEED_ZLIB AND ZLIB_LIBRARY)
-      if(HDF5_NEED_SZIP AND SZIP_LIBRARY)
-        target_link_libraries(cgns_shared ${SZIP_LIBRARY})
-      endif(HDF5_NEED_SZIP AND SZIP_LIBRARY)
-      if(HDF5_NEED_MPI AND MPI_LIBS)
-        target_link_libraries(cgns_shared ${MPI_LIBS})
-      endif(HDF5_NEED_MPI AND MPI_LIBS)
-    endif (ENABLE_HDF5 AND HDF5_LIBRARY)
-  endif (WIN32 OR CYGWIN)
+  endif(WIN32 OR CYGWIN)
+
+  if (ENABLE_HDF5 AND HDF5_LIBRARY)
+    target_link_libraries(cgns_shared ${HDF5_LIBRARY})
+    if(HDF5_NEED_ZLIB AND ZLIB_LIBRARY)
+      target_link_libraries(cgns_shared ${ZLIB_LIBRARY})
+    endif(HDF5_NEED_ZLIB AND ZLIB_LIBRARY)
+    if(HDF5_NEED_SZIP AND SZIP_LIBRARY)
+      target_link_libraries(cgns_shared ${SZIP_LIBRARY})
+    endif(HDF5_NEED_SZIP AND SZIP_LIBRARY)
+    if(HDF5_NEED_MPI AND MPI_LIBRARY)
+      target_link_libraries(cgns_shared ${MPI_LIBRARY})
+    endif(HDF5_NEED_MPI AND MPI_LIBRARY)
+  endif (ENABLE_HDF5 AND HDF5_LIBRARY)
 else(CGNS_BUILD_SHARED)
   set(CGNS_USE_SHARED "OFF")
   mark_as_advanced(FORCE CGNS_USE_SHARED)
 endif(CGNS_BUILD_SHARED)
 
 # Change the output name of the library to be libcgns
-set_target_properties(cgns_static PROPERTIES OUTPUT_NAME cgns)
+set_target_properties(cgns_static PROPERTIES OUTPUT_NAME cgns.${CGNS_VERSION})
 set_target_properties(cgns_static PROPERTIES CLEAN_DIRECT_OUTPUT 1)
 if(CGNS_BUILD_SHARED)
   # for windows we need to change the name of the shared library
   # for both static and shared version to exist
   if (WIN32 OR CYGWIN)
-    set_target_properties(cgns_shared PROPERTIES OUTPUT_NAME cgnsdll)
+    set_target_properties(cgns_shared PROPERTIES OUTPUT_NAME cgnsdll.${CGNS_VERSION})
   else (WIN32 OR CYGWIN)
     set_target_properties(cgns_shared PROPERTIES OUTPUT_NAME cgns)
   endif (WIN32 OR CYGWIN)
@@ -244,10 +245,10 @@
 if(CGNS_BUILD_SHARED)
   # for windows, need to install both cgnsdll.dll and cgnsdll.lib
   if (WIN32 OR CYGWIN)
-    install(TARGETS cgns_shared ARCHIVE DESTINATION lib)
-    install(TARGETS cgns_shared RUNTIME DESTINATION lib)
+    install(TARGETS cgns_shared ARCHIVE DESTINATION ${LIB_SUFFIX})
+    install(TARGETS cgns_shared RUNTIME DESTINATION bin)
   else (WIN32 OR CYGWIN)
-    install(TARGETS cgns_shared LIBRARY DESTINATION lib)
+    install(TARGETS cgns_shared LIBRARY DESTINATION ${LIB_SUFFIX})
   endif (WIN32 OR CYGWIN)
 endif(CGNS_BUILD_SHARED)
 
@@ -260,12 +261,12 @@
 	${CMAKE_CURRENT_BINARY_DIR}/cgnstypes_f.h
 	${CMAKE_CURRENT_BINARY_DIR}/cgnslib_f.h
	${CMAKE_CURRENT_BINARY_DIR}/cgnsBuild.defs
-	DESTINATION include)
+	DESTINATION include/cgns-${CGNS_VERSION})
 
 if (ENABLE_LEGACY)
-  install(FILES adf/ADF.h DESTINATION include/adf)
+  install(FILES adf/ADF.h DESTINATION include/cgns-${CGNS_VERSION}/adf)
   if (ENABLE_HDF5)
-    install(FILES adfh/ADFH.h DESTINATION include/adfh)
+    install(FILES adfh/ADFH.h DESTINATION include/cgns-${CGNS_VERSION}/adfh)
   endif (ENABLE_HDF5)
 endif (ENABLE_LEGACY)
 
