--- src/cgnstools/common/tree.tcl.orig	2014-03-10 12:14:30.605522705 +0100
+++ src/cgnstools/common/tree.tcl	2014-03-10 12:15:35.306522694 +0100
@@ -25,6 +25,22 @@
    0x11, 0x01, 0x7d, 0x01, 0x11, 0x01,
    0x11, 0x01, 0x01, 0x01, 0xff, 0x01};}
 
+proc file2 {op v} {
+  set dircomps [split $v "/"]
+
+  if {$op == "dirname"} {
+    set dir [join [lrange $dircomps 0 end-1] /]
+    if {$dir == ""} {
+      set dir "/"
+    }
+    return $dir
+  }
+  if {$op == "tail"} {
+    set n [lindex $dircomps end]
+    return $n
+  }
+}
+
 #----- create a new tree widget
 
 proc TreeCreate {w args} {
@@ -109,8 +125,8 @@
   if {$v == "/"} {
     set _Tree($w:root) 1
   } else {
-    set dir [file dirname $v]
-    set n [file tail $v]
+    set dir [file2 dirname $v]
+    set n [file2 tail $v]
     foreach {op arg} $args {
       switch -exact -- $op {
         -dir {
@@ -174,8 +190,8 @@
   foreach t [array names _Tree $w:$v:*] {
     unset _Tree($t)
   }
-  set dir [file dirname $v]
-  set n [file tail $v]
+  set dir [file2 dirname $v]
+  set n [file2 tail $v]
   set i [lsearch -exact $_Tree($w:$dir:children) $n]
   if {$i >= 0} {
     set _Tree($w:$dir:children) [lreplace $_Tree($w:$dir:children) $i $i]
@@ -192,14 +208,14 @@
 proc TreeMove {w vold vnew} {
   global _Tree
   if {![TreeExists $w $vold]} return
-  set pold [file dirname $vold]
-  set pnew [file dirname $vnew]
+  set pold [file2 dirname $vold]
+  set pnew [file2 dirname $vnew]
   if {$pnew == "."} {
-    if {$vnew == [file tail $vold]} return
+    if {$vnew == [file2 tail $vold]} return
   } else {
     if {![TreeExists $w $pnew]} return
   }
-  set n [lsearch -exact $_Tree($w:$pold:children) [file tail $vold]]
+  set n [lsearch -exact $_Tree($w:$pold:children) [file2 tail $vold]]
   if {$pnew == "."} {
     set _Tree($w:$pold:children) \
       [lreplace $_Tree($w:$pold:children) $n $n $vnew]
@@ -212,7 +228,7 @@
   } else {
     set _Tree($w:$pold:children) \
       [lreplace $_Tree($w:$pold:children) $n $n]
-    lappend _Tree($w:$pnew:children) [file tail $vnew]
+    lappend _Tree($w:$pnew:children) [file2 tail $vnew]
   }
   if {$_Tree($w:sort) != ""} {
     set _Tree($w:$pnew:children) [eval lsort $_Tree($w:sort) \
@@ -223,7 +239,7 @@
     set _Tree($w:$vnew[string range $told $n end]) $_Tree($told)
     unset _Tree($told)
   }
-  set _Tree($w:$vnew:text) [file tail $vnew]
+  set _Tree($w:$vnew:text) [file2 tail $vnew]
   Tree:buildwhenidle $w
 }
 
@@ -439,8 +455,8 @@
     if {$node != ""} {return $node}
   }
   while {$v != "/"} {
-    set parent [file dirname $v]
-    set node [file tail $v]
+    set parent [file2 dirname $v]
+    set node [file2 tail $v]
     set n [lsearch -exact $_Tree($w:$parent:children) $node]
     if {$n >= 0} {
       incr n
@@ -459,7 +475,7 @@
   if {![info exists _Tree($w:$v:open)]} return
   set redraw 0
   set node {}
-  foreach p [split [file dirname $v] /] {
+  foreach p [split [file2 dirname $v] /] {
     if {$p != {}} {
       append node "/$p"
       if {!$_Tree($w:$node:open)} {
@@ -516,7 +532,7 @@
     [$w cget -highlightthickness]}]
   set wmax [expr {[$w canvasx $wd] - $x}]
 
-  set _Tree($w:edit) [file tail $v]
+  set _Tree($w:edit) [file2 tail $v]
   set selection [TreeSelectionGet $w]
   TreeSelectionSet $w ""
   $w itemconfigure $tag -fill $bg
@@ -604,7 +620,7 @@
   if {$v == "/"} {
     set _Tree($w:$v:text) $v
   } else {
-    set _Tree($w:$v:text) [file tail $v]
+    set _Tree($w:$v:text) [file2 tail $v]
   }
 }
 
--- src/cgnstools/cgnsview/cgnsview.tcl.orig	2014-03-10 12:14:46.212522702 +0100
+++ src/cgnstools/cgnsview/cgnsview.tcl	2014-03-10 12:15:58.218522692 +0100
@@ -2,6 +2,23 @@
 # the next line restarts using wish \
 exec cgiowish -f "$0" "$@"
 
+proc file2 {op v} {
+  set dircomps [split $v "/"]
+
+  if {$op == "dirname"} {
+    set dir [join [lrange $dircomps 0 end-1] /]
+    if {$dir == ""} {
+      set dir "/"
+    }
+    return $dir
+  }
+  if {$op == "tail"} {
+    set n [lindex $dircomps end]
+    return $n
+  }
+}
+
+
 proc error_exit {msg} {
   wm withdraw .
   tk_dialog .error Error $msg error 0 Exit
@@ -1327,7 +1344,7 @@
 
 proc check_links {node} {
   set path {}
-  foreach p [split [file dirname $node] /] {
+  foreach p [split [file2 dirname $node] /] {
     if {$p != ""} {
       append path "/$p"
       if {[CGIOlink $path] != {}} {return 1}
@@ -1408,8 +1425,8 @@
       set Node(parent) {}
       set Node(name) $node
     } else {
-      set Node(parent) [file dirname $node]
-      set Node(name) [file tail $node]
+      set Node(parent) [file2 dirname $node]
+      set Node(name) [file2 tail $node]
     }
     catch {CGIOlabel $node} Node(label)
     if {![catch {CGIOtype $node} type]} {
@@ -1910,13 +1927,13 @@
   set node [TreeSelectionGet $w]
   if {$node == ""} return
   set name [TreeEdit $w $node]
-  if {$name == "" || $name == [file tail $node]} return
+  if {$name == "" || $name == [file2 tail $node]} return
   do_backup
   if {[catch {CGIOname $node $name} msg]} {
     errormsg $msg
   } else {
     TreeMove $w $node $name
-    set node [get_node [file dirname $node] $name]
+    set node [get_node [file2 dirname $node] $name]
   }
   file_reload
   if {[CGIOnode $node] != ""} {
--- src/cgnstools/cgnsplot/cgnsplot.tcl.orig	2014-03-10 12:15:00.898522699 +0100
+++ src/cgnstools/cgnsplot/cgnsplot.tcl	2014-03-10 12:16:23.980522688 +0100
@@ -2,6 +2,23 @@
 # the next line restarts using wish \
 exec plotwish -f "$0" "$@"
 
+proc file2 {op v} {
+  set dircomps [split $v "/"]
+
+  if {$op == "dirname"} {
+    set dir [join [lrange $dircomps 0 end-1] /]
+    if {$dir == ""} {
+      set dir "/"
+    }
+    return $dir
+  }
+  if {$op == "tail"} {
+    set n [lindex $dircomps end]
+    return $n
+  }
+}
+
+
 proc error_exit {msg} {
   wm withdraw .
   tk_dialog .error Error $msg error 0 Exit
@@ -1341,7 +1358,7 @@
         set ProgData($reg,clr) $clr
         set ProgData($reg,msg) [CGNSsummary $nz $nr]
       }
-      set dir [file dirname $r]
+      set dir [file2 dirname $r]
       if {$dir != "" && $dir != "."} {
         if {[lsearch $subdirs $dir] < 0} {
           lappend subdirs $dir
@@ -1480,7 +1497,7 @@
     TreeInsert $ProgData(tree) $zone -tag $nz -icon $icon -iconbg $clr
     set nr 0
     foreach r $ProgData($nz,regions) {
-      set dir [file dirname $r]
+      set dir [file2 dirname $r]
       if {$dir != "" && $dir != "." &&
         ![TreeExists $ProgData(tree) $zone/$dir]} {
         if {$ProgData($zone/$dir,dim) == 0} {
@@ -1640,10 +1657,10 @@
 proc node_visible {node} {
   global ProgData
   if {$ProgData($node,vis) == 0} {return 0}
-  set node [file dirname $node]
+  set node [file2 dirname $node]
   while {$node != "" && $node != "." && $node != "/"} {
     if {$ProgData($node,vis) == 0} {return 0}
-    set node [file dirname $node]
+    set node [file2 dirname $node]
   }
   return 1
 }
@@ -1710,7 +1727,7 @@
   if {$mode == ""} return
   set ProgData($node,mode) $mode
   if {$node != "/"} {
-    parent_mode [file dirname $node]
+    parent_mode [file2 dirname $node]
   }
 }
 
@@ -1720,7 +1737,7 @@
   if {$node == "" || $ProgData($node,dim) == 0} return
   update_node $node $ProgData(curmode)
   if {$node != "/"} {
-    parent_mode [file dirname $node]
+    parent_mode [file2 dirname $node]
   }
 }
 
