--- cgnslib.h.orig	2011-02-11 23:29:29.000000000 +0100
+++ cgnslib.h	2014-02-06 17:22:00.266197873 +0100
@@ -27,9 +27,9 @@
 #ifndef CGNSDLL
 # ifdef _WIN32
 #  if defined(BUILD_DLL)
-#    define CGNSDLL _declspec(dllexport)
+#    define CGNSDLL __declspec(dllexport)
 #  elif defined(USE_DLL)
-#    define CGNSDLL _declspec(dllimport)
+#    define CGNSDLL __declspec(dllimport)
 #  else
 #    define CGNSDLL
 #  endif
--- cgns_header.h.orig	2009-08-20 20:10:21.000000000 +0200
+++ cgns_header.h	2014-02-06 17:23:53.355197856 +0100
@@ -909,7 +909,7 @@
 /* need some of these routines exported for CGNStools */
 
 #if defined(_WIN32) && defined(BUILD_DLL)
-# define CGNSDLL _declspec(dllexport)
+# define CGNSDLL __declspec(dllexport)
 #else
 # define CGNSDLL
 #endif
--- adf_cond.c.orig	2007-01-01 00:53:14.000000000 +0100
+++ adf_cond.c	2014-02-06 17:23:38.962197858 +0100
@@ -28,7 +28,7 @@
 #endif
 
 #if defined(_WIN32) && defined(BUILD_DLL)
-# define CGNSDLL _declspec(dllexport)
+# define CGNSDLL __declspec(dllexport)
 #else
 # define CGNSDLL
 #endif
--- adf/ADF.h.orig	2007-01-01 00:53:15.000000000 +0100
+++ adf/ADF.h	2014-02-06 17:23:02.762197864 +0100
@@ -16,7 +16,7 @@
 #define ADF_INCLUDE
 
 #if defined(_WIN32) && defined(BUILD_DLL)
-# define EXTERN extern _declspec(dllexport)
+# define EXTERN extern __declspec(dllexport)
 #else
 # define EXTERN extern
 #endif
--- adfh/ADFH.h.orig	2009-08-31 23:56:39.000000000 +0200
+++ adfh/ADFH.h	2014-02-06 17:23:18.010197861 +0100
@@ -137,7 +137,7 @@
 ***********************************************************************/
 
 #if defined(_WIN32) && defined(BUILD_DLL)
-# define EXTERN extern _declspec(dllexport)
+# define EXTERN extern __declspec(dllexport)
 #else
 # define EXTERN extern
 #endif
