--- CMakeLists.txt	2013-01-08 16:56:07.633928286 +0100
+++ CMakeLists.txt	2013-01-08 16:18:26.750849891 +0100
@@ -1,5 +1,42 @@
 cmake_minimum_required(VERSION 2.8)
-project(SuiteSparse-4.2.1 C)
+project(SuiteSparse-4.2.1 Fortran C)
+
+IF(NOT CMAKE_CROSSCOMPILING AND NOT CFS_FORTRAN_COMPILER_NAME STREQUAL "OPEN64")
+  # Create a header file blas_lapack.h with the correct mangling
+  # for the Fortran routines called in my C programs.
+  # cf. http://www.netlib.org/lapack/lawnspdf/lawn270.pdf
+  include(FortranCInterface)
+  FortranCInterface_HEADER(blas_lapack.h
+    MACRO_NAMESPACE "BLAS_LAPACK_"
+    SYMBOLS 
+    dtrsv
+    dgemv
+    dtrsm
+    dgemm
+    dsyrk
+    dger
+    dscal
+    dpotrf
+
+    ztrsv
+    zgemv
+    ztrsm
+    zgemm
+    zherk
+    zgeru
+    zscal
+    zpotrf
+  )
+
+  # To find the newly generated header file blas_lapack.h at compile time
+  include_directories (${CMAKE_CURRENT_BINARY_DIR})
+ELSE()
+  include_directories (${CMAKE_CURRENT_SOURCE_DIR})
+ENDIF()
+
+IF(NOT MSVC)
+  ADD_DEFINITIONS(-DBLAS_NO_UNDERSCORE)
+ENDIF()
 
 add_subdirectory(UFconfig)
 add_subdirectory(AMD)
@@ -9,6 +41,7 @@
 add_subdirectory(CHOLMOD)
 add_subdirectory(UMFPACK)
 
+
 # add_library(libsuitesparse SHARED libsuitesparse.def)
 # target_link_libraries(libsuitesparse
 #    libufconfig libamd libcamd libcolamd libccolamd libcholmod libumfpack)
--- CHOLMOD/Include/cholmod_blas.h	2009-11-07 14:31:40.000000000 +0100
+++ CHOLMOD/Include/cholmod_blas.h	2013-01-08 16:12:02.718849958 +0100
@@ -105,6 +105,8 @@
 
 #elif defined (BLAS_NO_UNDERSCORE)
 
+#include <blas_lapack.h>
+
 #define BLAS_DTRSV dtrsv
 #define BLAS_DGEMV dgemv
 #define BLAS_DTRSM dtrsm
--- UMFPACK/Source/cholmod_blas.h	2009-11-07 14:31:04.000000000 +0100
+++ UMFPACK/Source/cholmod_blas.h	2013-01-08 15:59:51.358916436 +0100
@@ -97,6 +97,8 @@
 
 #elif defined (BLAS_NO_UNDERSCORE)
 
+#include <blas_lapack.h>
+
 #define BLAS_DTRSV dtrsv
 #define BLAS_DGEMV dgemv
 #define BLAS_DTRSM dtrsm
