--- configure.orig	2012-11-17 17:04:08.000000000 +0100
+++ configure	2012-11-17 17:05:12.000000000 +0100
@@ -435,7 +435,7 @@
         echo "Type 'c' if you want to use the Commercial Edition."
         echo "Type 'o' if you want to use the Open Source Edition."
         echo
-        read commercial
+        commercial="o"
         echo
         if [ "$commercial" = "c" ]; then
             COMMERCIAL_USER="yes"
@@ -4545,7 +4545,7 @@
             echo "Type 'no' to decline this license offer."
             echo
             echo $ECHO_N "Do you accept the terms of $affix license? $ECHO_C"
-            read acceptance
+            acceptance="yes"
         fi
         echo
         if [ "$acceptance" = "yes" ] || [ "$acceptance" = "y" ]; then
