--- src/muParser.cpp.orig	2011-11-24 23:27:12.341989187 +0100
+++ src/muParser.cpp	2011-11-24 23:40:39.285514544 +0100
@@ -37,6 +37,11 @@
 /** \brief The eulerian number. */
 #define PARSER_CONST_E   2.718281828459045235360287
 
+#ifdef __INTEL_COMPILER 
+/** @brief Replacement of unknown atomic function. Bug report submitted to Intel. */
+#define __sync_fetch_and_add(ptr,addend) _InterlockedExchangeAdd(const_cast<void*>(reinterpret_cast<volatile void*>(ptr)), addend);
+#endif
+
 using namespace std;
 
 /** \file
