--- UTIL/second.f.orig	2012-06-04 19:30:57.000000000 +0200
+++ UTIL/second.f	2012-06-04 19:32:07.000000000 +0200
@@ -13,21 +13,7 @@
 *  SECOND returns the user time for a process in seconds.
 *  This version gets the time from the system function ETIME.
 *
-*     .. Local Scalars ..
-      REAL               T1
-*     ..
-*     .. Local Arrays ..
-      REAL               TARRAY( 2 )
-*     ..
-*     .. External Functions ..
-      REAL               ETIME
-      INTRINSIC          ETIME
-*     ..
-*     .. Executable Statements ..
-*
-
-      T1 = ETIME( TARRAY )
-      T  = TARRAY( 1 )
+      CALL  cpu_time(t)
 
       RETURN
 *
