--- src/CMakeLists.txt	2011-11-07 23:11:30.000000000 +0100
+++ src/CMakeLists.txt	2012-11-26 18:33:46.057764985 +0100
@@ -570,27 +570,27 @@
   TARGET_LINK_LIBRARIES (H5detect "ws2_32.lib")
 ENDIF (MSVC)
 
-SET (CMD $<TARGET_FILE:H5detect>)
-ADD_CUSTOM_COMMAND (
-    OUTPUT ${HDF5_BINARY_DIR}/H5Tinit.c
-    COMMAND ${CMD}
-    ARGS > ${HDF5_BINARY_DIR}/H5Tinit.c
-    DEPENDS H5detect
-)
+# SET (CMD $<TARGET_FILE:H5detect>)
+# ADD_CUSTOM_COMMAND (
+#     OUTPUT ${HDF5_BINARY_DIR}/H5Tinit.c
+#     COMMAND ${CMD}
+#     ARGS > ${HDF5_BINARY_DIR}/H5Tinit.c
+#     DEPENDS H5detect
+# )
 
 ADD_EXECUTABLE (H5make_libsettings ${HDF5_SRC_DIR}/H5make_libsettings.c)
 IF (MSVC)
   TARGET_LINK_LIBRARIES (H5make_libsettings "ws2_32.lib")
 ENDIF (MSVC)
 
-SET (CMD $<TARGET_FILE:H5make_libsettings>)
-ADD_CUSTOM_COMMAND (
-    OUTPUT ${HDF5_BINARY_DIR}/H5lib_settings.c
-    COMMAND ${CMD}
-    ARGS > ${HDF5_BINARY_DIR}/H5lib_settings.c
-    DEPENDS H5make_libsettings
-    WORKING_DIRECTORY ${HDF5_BINARY_DIR}
-)
+# SET (CMD $<TARGET_FILE:H5make_libsettings>)
+# ADD_CUSTOM_COMMAND (
+#     OUTPUT ${HDF5_BINARY_DIR}/H5lib_settings.c
+#     COMMAND ${CMD}
+#     ARGS > ${HDF5_BINARY_DIR}/H5lib_settings.c
+#     DEPENDS H5make_libsettings
+#     WORKING_DIRECTORY ${HDF5_BINARY_DIR}
+# )
 
 ADD_CUSTOM_COMMAND (
     TARGET ${HDF5_BINARY_DIR}/H5Edefin.h
