--- SuperBuild/External_Manta.cmake.orig	2012-11-11 06:03:17.031693931 +0100
+++ SuperBuild/External_Manta.cmake	2012-11-11 06:04:30.035689208 +0100
@@ -15,7 +15,7 @@
 set(manta_PATCH_COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/manta_patch_step.cmake)
 
 ExternalProject_Add(Manta
-  DOWNLOAD_DIR ${CMAKE_CURRENT_BINARY_DIR}
+  DOWNLOAD_DIR ${CFS_DEPS_CACHE_DIR}/sources/paraview 
   SOURCE_DIR ${Manta_source}
   BINARY_DIR ${Manta_binary}
   URL ${MANTA_URL}/${MANTA_GZ} 
