--- SuperBuild/External_freetype.cmake.orig	2012-11-13 19:13:29.983525518 +0100
+++ SuperBuild/External_freetype.cmake	2012-11-13 19:14:01.968568930 +0100
@@ -25,7 +25,7 @@
   set(freetype_BUILD_COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/freetype_build_step.cmake)
 
   ExternalProject_Add(freetype
-    DOWNLOAD_DIR ${CMAKE_CURRENT_BINARY_DIR}
+    DOWNLOAD_DIR ${CFS_DEPS_CACHE_DIR}/sources/paraview
     SOURCE_DIR ${freetype_source}
     INSTALL_DIR ${freetype_install}
     URL ${FT_URL}/${FT_GZ}
@@ -57,7 +57,7 @@
 else()
 
   ExternalProject_Add(freetype
-    DOWNLOAD_DIR ${CMAKE_CURRENT_BINARY_DIR}
+    DOWNLOAD_DIR ${CFS_DEPS_CACHE_DIR}/sources/paraview
     SOURCE_DIR ${freetype_source}
     INSTALL_DIR ${freetype_install}
     URL ${FT_URL}/${FT_GZ}
