--- SuperBuild/External_MPICH2.cmake.orig	2012-11-11 05:51:06.893733693 +0100
+++ SuperBuild/External_MPICH2.cmake	2012-11-11 05:53:16.132725070 +0100
@@ -9,7 +9,7 @@
   
   message("Fatal Error. Don't Build MPICH2 on Windows, it requires a service!")
   ExternalProject_Add(MPICH2
-    DOWNLOAD_DIR ${CMAKE_CURRENT_BINARY_DIR}
+    DOWNLOAD_DIR ${CFS_DEPS_CACHE_DIR}/sources/paraview 
     SOURCE_DIR ${MPICH2_source}
     BINARY_DIR ${MPICH2_build}
     INSTALL_DIR ${MPICH2_install}
@@ -31,7 +31,7 @@
   # on linux build the binaries static
   ExternalProject_Add(MPICH2
     LIST_SEPARATOR ^^
-    DOWNLOAD_DIR ${CMAKE_CURRENT_BINARY_DIR}
+    DOWNLOAD_DIR ${CFS_DEPS_CACHE_DIR}/sources/paraview
     SOURCE_DIR ${MPICH2_source}
     INSTALL_DIR ${MPICH2_install}
     URL ${MPICH2_URL}/${MPICH2_GZ}