--- SuperBuild/External_Boost.cmake.orig	2012-11-11 00:27:37.512779842 +0100
+++ SuperBuild/External_Boost.cmake	2012-11-11 00:30:42.836770392 +0100
@@ -16,11 +16,18 @@
   )
 endif()
 
+configure_file(${ParaViewSuperBuild_CMAKE_SOURCE_DIR}/boost_patch_step.cmake.in
+  ${CMAKE_CURRENT_BINARY_DIR}/boost_patch_step.cmake
+  @ONLY)
+
+set(Boost_PATCH_COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/boost_patch_step.cmake)
+
 ExternalProject_Add(Boost
-  DOWNLOAD_DIR ${CMAKE_CURRENT_BINARY_DIR}
+  DOWNLOAD_DIR ${CFS_DEPS_CACHE_DIR}/sources/boost
   URL ${BOOST_URL}/${BOOST_GZ}
   URL_MD5 ${BOOST_MD5}
   UPDATE_COMMAND ""
+  PATCH_COMMAND ${Boost_PATCH_COMMAND}
   SOURCE_DIR ${boost_source}
   BINARY_DIR ${boost_binary}
   CMAKE_CACHE_ARGS