--- SuperBuild/External_CGNS.cmake.orig	2012-11-11 05:27:36.474810794 +0100
+++ SuperBuild/External_CGNS.cmake	2012-11-11 05:29:27.514802279 +0100
@@ -38,6 +38,7 @@
     OUTPUT_STRIP_TRAILING_WHITESPACE)
 
   ExternalProject_Add(CGNS
+    DOWNLOAD_DIR ${CFS_DEPS_CACHE_DIR}/sources/cgns
     URL ${CGNS_URL}/${CGNS_GZ}
     URL_MD5 ${CGNS_MD5}
     SOURCE_DIR ${CGNS_source}
@@ -69,6 +70,7 @@
   set(CGNS_CONFIGURE_COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/CGNS_configure_step.cmake)
 
   ExternalProject_Add(CGNS
+    DOWNLOAD_DIR ${CFS_DEPS_CACHE_DIR}/sources/cgns
     SOURCE_DIR ${CGNS_source}
     INSTALL_DIR ${CGNS_install}
     URL ${CGNS_URL}/${CGNS_GZ}
@@ -86,6 +88,7 @@
   file(MAKE_DIRECTORY "${CGNS_install}/include")
 
   ExternalProject_Add(CGNS
+    DOWNLOAD_DIR ${CFS_DEPS_CACHE_DIR}/sources/cgns
     SOURCE_DIR ${CGNS_source}
     INSTALL_DIR ${CGNS_install}
     URL ${CGNS_URL}/${CGNS_GZ}