--- SuperBuild/External_Silo.cmake.orig	2012-11-11 11:45:12.799584895 +0100
+++ SuperBuild/External_Silo.cmake	2012-11-11 11:46:13.679581257 +0100
@@ -33,7 +33,7 @@
   set(Silo_INSTALL_COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/Silo_install_step.cmake)
 
   ExternalProject_Add(Silo
-    DOWNLOAD_DIR ${CMAKE_CURRENT_BINARY_DIR}
+    DOWNLOAD_DIR ${CFS_DEPS_CACHE_DIR}/sources/paraview
     SOURCE_DIR ${Silo_source}
     INSTALL_DIR ${Silo_install}
     URL ${SILO_URL}/${SILO_GZ}
@@ -81,7 +81,7 @@
   set(Silo_CONFIGURE_COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/silo_configure_step.cmake)
 
   ExternalProject_Add(Silo
-    DOWNLOAD_DIR ${CMAKE_CURRENT_BINARY_DIR}
+    DOWNLOAD_DIR ${CFS_DEPS_CACHE_DIR}/sources/paraview
     SOURCE_DIR ${Silo_source}
     INSTALL_DIR ${Silo_install}
     URL ${SILO_URL}/${SILO_GZ}