--- SuperBuild/External_HDF5.cmake.orig	2012-11-11 01:01:29.069670951 +0100
+++ SuperBuild/External_HDF5.cmake	2012-11-11 01:02:31.415662594 +0100
@@ -6,6 +6,7 @@
 set(HDF5_binary "${CMAKE_CURRENT_BINARY_DIR}/HDF5-build")

 ExternalProject_Add(HDF5
+  DOWNLOAD_DIR ${CFS_DEPS_CACHE_DIR}/sources/hdf5
   URL ${HDF5_URL}/${HDF5_GZ}
   URL_MD5 ${HDF5_MD5}
   UPDATE_COMMAND ""
@@ -20,6 +21,7 @@
     -DCMAKE_BUILD_TYPE:STRING=${CMAKE_CFG_INTDIR}
     ${pv_tpl_compiler_args}
     -DHDF5_ENABLE_Z_LIB_SUPPORT:BOOL=ON
+    -DHDF5_BUILD_CPP_LIB:BOOL=ON
     -DHDF5_BUILD_HL_LIB:BOOL=ON
     -DHDF5_BUILD_WITH_INSTALL_NAME:BOOL=ON
     -DZLIB_INCLUDE_DIR:STRING=${ZLIB_INCLUDE_DIR}
@@ -33,9 +35,11 @@
 if(WIN32)
   set(HDF5_INCLUDE_DIR ${HDF5_install}/include)
   set(HDF5_LIBRARY ${HDF5_install}/lib/hdf5dll${_LINK_LIBRARY_SUFFIX})
+  set(HDF5_CPP_LIBRARY ${HDF5_install}/lib/hdf5_cppdll${_LINK_LIBRARY_SUFFIX})
   set(HDF5_HL_LIBRARY ${HDF5_install}/lib/hdf5_hldll${_LINK_LIBRARY_SUFFIX})
 else()
   set(HDF5_INCLUDE_DIR ${HDF5_install}/include)
   set(HDF5_LIBRARY ${HDF5_install}/lib/libhdf5${_LINK_LIBRARY_SUFFIX})
+  set(HDF5_CPP_LIBRARY ${HDF5_install}/lib/libhdf5_cpp${_LINK_LIBRARY_SUFFIX})
   set(HDF5_HL_LIBRARY ${HDF5_install}/lib/libhdf5_hl${_LINK_LIBRARY_SUFFIX})
 endif()
