--- Qt/Components/pqColorScaleEditor.cxx.orig	2012-04-10 12:17:13.000000000 +0200
+++ Qt/Components/pqColorScaleEditor.cxx	2012-04-10 12:21:45.000000000 +0200
@@ -1354,6 +1354,57 @@
   colorMap.addPoint(pqChartValue((double)1.0),  QColor(224,   0, 255 ), (double)1.0);
   colorMap.setNanColor(QColor(255, 255, 0));
   model->addBuiltinColorMap(colorMap, "Rainbow Blended Grey");
+
+  colorMap.removeAllPoints();
+  colorMap.setColorSpace(pqColorMapModel::RgbSpace);
+  colorMap.addPoint(pqChartValue((double)0.0), QColor(0,     0,   127  ), (double)0.0);
+  colorMap.addPoint(pqChartValue((double)0.125), QColor(0,     0,   255  ), (double)0.125);
+  colorMap.addPoint(pqChartValue((double)0.375), QColor(0,   255,   255  ), (double)0.375);
+  colorMap.addPoint(pqChartValue((double)0.625), QColor(255, 255,   0    ), (double)0.625);
+  colorMap.addPoint(pqChartValue((double)0.875), QColor(255,   0,   0    ), (double)0.875);
+  colorMap.addPoint(pqChartValue((double)1.0), QColor(127,   0,   0    ), (double)1.0);
+  model->addBuiltinColorMap(colorMap, "Matlab (Jet)");
+
+  colorMap.removeAllPoints();
+  colorMap.setColorSpace(pqColorMapModel::HsvSpace);
+  colorMap.addPoint(pqChartValue((double)0.0), QColor(0,     0,   127  ), (double)0.0);
+  colorMap.addPoint(pqChartValue((double)0.125), QColor(0,     0,   255  ), (double)0.125);
+  colorMap.addPoint(pqChartValue((double)0.375), QColor(0,   255,   255  ), (double)0.375);
+  colorMap.addPoint(pqChartValue((double)0.625), QColor(255, 255,   0    ), (double)0.625);
+  colorMap.addPoint(pqChartValue((double)0.875), QColor(255,   0,   0    ), (double)0.875);
+  colorMap.addPoint(pqChartValue((double)1.0), QColor(127,   0,   0    ), (double)1.0);
+  model->addBuiltinColorMap(colorMap, "GiD");
+
+  colorMap.removeAllPoints();
+  colorMap.setColorSpace(pqColorMapModel::RgbSpace);
+  colorMap.addPoint(pqChartValue((double)0.0), QColor(212,     255,   212  ), (double)0.0);
+  colorMap.addPoint(pqChartValue((double)0.25), QColor(0,     255,   0  ), (double)0.25);
+  colorMap.addPoint(pqChartValue((double)0.5), QColor(64,   64,   64  ), (double)0.5);
+  colorMap.addPoint(pqChartValue((double)0.75), QColor(255, 0,   0    ), (double)0.75);
+  colorMap.addPoint(pqChartValue((double)1.0), QColor(255,   212,   212    ), (double)1.0);
+  model->addBuiltinColorMap(colorMap, "Polytec");
+
+  colorMap.removeAllPoints();
+  colorMap.setColorSpace(pqColorMapModel::RgbSpace);
+  colorMap.addPoint(pqChartValue((double)0.0), QColor(0,     0,   128  ), (double)0.0);
+  colorMap.addPoint(pqChartValue((double)0.1), QColor(0,     0,   255  ), (double)0.1);
+  colorMap.addPoint(pqChartValue((double)0.2), QColor(0,   255,   255  ), (double)0.2);
+  colorMap.addPoint(pqChartValue((double)0.32), QColor(0,  204,   128    ), (double)0.32);
+  colorMap.addPoint(pqChartValue((double)0.42), QColor(0,  255,   0    ), (double)0.42);
+  colorMap.addPoint(pqChartValue((double)0.52), QColor(255,  255,   0    ), (double)0.52);
+  colorMap.addPoint(pqChartValue((double)0.75), QColor(255,  0,   0    ), (double)0.75);
+  colorMap.addPoint(pqChartValue((double)0.85), QColor(255,  0,   255    ), (double)0.85);
+  colorMap.addPoint(pqChartValue((double)1.0), QColor(255,  204,   255    ), (double)1.0);
+  model->addBuiltinColorMap(colorMap, "CapaPost");
+
+  colorMap.removeAllPoints();
+  colorMap.setColorSpace(pqColorMapModel::RgbSpace);
+  colorMap.addPoint(pqChartValue((double)0.0), QColor(0,     0,   128  ), (double)0.0);
+  colorMap.addPoint(pqChartValue((double)0.4375), QColor(255,     0,   128  ), (double)0.4375);
+  colorMap.addPoint(pqChartValue((double)0.5625), QColor(255,     0,   0  ), (double)0.5625);
+  colorMap.addPoint(pqChartValue((double)1.0), QColor(255,  255,   166    ), (double)1.0);
+  model->addBuiltinColorMap(colorMap, "Hot Cold");
+
 }
 
 void pqColorScaleEditor::loadColorPoints()
