--- VTK/IO/vtkSQLDatabaseSchema.h.orig	2011-04-19 18:40:17.000000000 +0200
+++ VTK/IO/vtkSQLDatabaseSchema.h	2011-04-19 18:40:46.000000000 +0200
@@ -75,8 +75,8 @@
       REAL      = 6,
       DOUBLE    = 7,
       BLOB      = 8,
-      TIME      = 9,
-      DATE      = 10,
+      TIME_T    = 9,
+      DATE_D    = 10,
       TIMESTAMP = 11
     };
 
--- VTK/IO/vtkSQLDatabase.cxx.orig	2011-04-19 18:44:12.000000000 +0200
+++ VTK/IO/vtkSQLDatabase.cxx	2011-04-19 18:44:44.000000000 +0200
@@ -179,10 +179,10 @@
     case vtkSQLDatabaseSchema::BLOB:
       colTypeStr = "";
       break;
-    case vtkSQLDatabaseSchema::TIME:
+    case vtkSQLDatabaseSchema::TIME_T:
       colTypeStr = "TIME";
       break;
-    case vtkSQLDatabaseSchema::DATE:
+    case vtkSQLDatabaseSchema::DATE_D:
       colTypeStr = "DATE";
       break;
     case vtkSQLDatabaseSchema::TIMESTAMP:
@@ -231,10 +231,10 @@
     case vtkSQLDatabaseSchema::BLOB:
       colSizeType =  0;
       break;
-    case vtkSQLDatabaseSchema::TIME:
+    case vtkSQLDatabaseSchema::TIME_T:
       colSizeType =  0;
       break;
-    case vtkSQLDatabaseSchema::DATE:
+    case vtkSQLDatabaseSchema::DATE_D:
       colSizeType =  0;
       break;
     case vtkSQLDatabaseSchema::TIMESTAMP:
--- VTK/IO/vtkSQLiteDatabase.cxx.orig	2011-04-19 18:45:06.000000000 +0200
+++ VTK/IO/vtkSQLiteDatabase.cxx	2011-04-19 18:45:40.000000000 +0200
@@ -125,10 +125,10 @@
     case vtkSQLDatabaseSchema::BLOB:
       colTypeStr = "BLOB";
       break;
-    case vtkSQLDatabaseSchema::TIME:
+    case vtkSQLDatabaseSchema::TIME_T:
       colTypeStr = "TIME";
       break;
-    case vtkSQLDatabaseSchema::DATE:
+    case vtkSQLDatabaseSchema::DATE_D:
       colTypeStr = "DATE";
       break;
     case vtkSQLDatabaseSchema::TIMESTAMP:
@@ -176,10 +176,10 @@
     case vtkSQLDatabaseSchema::BLOB:
       colSizeType =  0;
       break;
-    case vtkSQLDatabaseSchema::TIME:
+    case vtkSQLDatabaseSchema::TIME_T:
       colSizeType =  0;
       break;
-    case vtkSQLDatabaseSchema::DATE:
+    case vtkSQLDatabaseSchema::DATE_D:
       colSizeType =  0;
       break;
     case vtkSQLDatabaseSchema::TIMESTAMP:
