--- CMakeLists.txt.orig	2013-01-08 20:20:25.698969842 +0100
+++ CMakeLists.txt	2013-01-08 20:21:08.193917415 +0100
@@ -44,12 +44,12 @@
 
 macro(lapack_install_library lib)
   install(TARGETS ${lib} EXPORT lapack-targets
-    ARCHIVE DESTINATION lib${LIB_SUFFIX}
-    LIBRARY DESTINATION lib${LIB_SUFFIX}
-    RUNTIME DESTINATION bin
+    ARCHIVE DESTINATION ${LIB_SUFFIX}/${CFS_ARCH_STR}
+    LIBRARY DESTINATION ${LIB_SUFFIX}/${CFS_ARCH_STR}
+    RUNTIME DESTINATION bin/${CFS_ARCH_STR}
   )
   if(GNUtoMS)
-    install(FILES ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/lib${lib}.lib DESTINATION lib)
+    install(FILES ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/lib${lib}.lib DESTINATION ${LIB_SUFFIX}/${CFS_ARCH_STR})
   endif()
 endmacro()
 
@@ -87,7 +87,7 @@
 set(SECOND_SRC  ${LAPACK_SOURCE_DIR}/INSTALL/second_${TIME_FUNC}.f)
 set(DSECOND_SRC  ${LAPACK_SOURCE_DIR}/INSTALL/dsecnd_${TIME_FUNC}.f)
 set(prefix ${CMAKE_INSTALL_PREFIX})
-set(libdir ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX})
+set(libdir ${CMAKE_INSTALL_PREFIX}/${LIB_SUFFIX}/${CFS_ARCH_STR})
 set(PKG_CONFIG_DIR ${libdir}/pkgconfig)
 
 # --------------------------------------------------
@@ -282,8 +282,8 @@
   ${LAPACK_GNUtoMS_IMPORT}
   ${LAPACK_BINARY_DIR}/CMakeFiles/lapack-config.cmake
   ${LAPACK_BINARY_DIR}/lapack-config-version.cmake
-  DESTINATION lib/cmake/lapack-${LAPACK_VERSION}
+  DESTINATION ${LIB_SUFFIX}/${CFS_ARCH_STR}/cmake/lapack-${LAPACK_VERSION}
   )
 
 install(EXPORT lapack-targets
-  DESTINATION lib/cmake/lapack-${LAPACK_VERSION})
+  DESTINATION ${LIB_SUFFIX}/${CFS_ARCH_STR}/cmake/lapack-${LAPACK_VERSION})
