--- zlib-1.2.6/CMakeLists.txt	2012-01-17 03:51:23.000000000 +0100
+++ CMakeLists.txt	2012-05-08 16:16:45.000000000 +0200
@@ -74,6 +74,7 @@
                ${CMAKE_CURRENT_BINARY_DIR}/zconf.h @ONLY)
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
+include_directories(${zlib_SOURCE_DIR})
 
 #============================================================================
 # zlib
@@ -165,9 +166,9 @@
 
 if(NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL )
     install(TARGETS zlib
-        RUNTIME DESTINATION bin
-        ARCHIVE DESTINATION lib
-        LIBRARY DESTINATION lib )
+        RUNTIME DESTINATION bin/${CFS_ARCH_STR}
+        ARCHIVE DESTINATION ${LIB_SUFFIX}/${CFS_ARCH_STR}
+        LIBRARY DESTINATION ${LIB_SUFFIX}/${CFS_ARCH_STR} )
 endif()
 if(NOT SKIP_INSTALL_HEADERS AND NOT SKIP_INSTALL_ALL )
     install(FILES ${ZLIB_PUBLIC_HDRS} DESTINATION include)
