--- CMakeLists.txt	2012-01-17 03:51:23.000000000 +0100
+++ CMakeLists.txt	2012-12-23 20:01:06.578379500 +0100
@@ -74,6 +74,7 @@
                ${CMAKE_CURRENT_BINARY_DIR}/zconf.h @ONLY)
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
+include_directories(${zlib_SOURCE_DIR})
 
 #============================================================================
 # zlib
@@ -129,13 +130,21 @@
         SET(CMAKE_RC_COMPILER windres.exe)
     endif()
 
+    SET(ARCH_TARGET "pe-i386")
+    IF(CFS_ARCH STREQUAL "X86_64")
+      SET(ARCH_TARGET "pe-x86-64")
+    ENDIF(CFS_ARCH STREQUAL "X86_64")
+    
     add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/zlib1rc.obj
                        COMMAND ${CMAKE_RC_COMPILER}
                             -D GCC_WINDRES
-                            -I ${CMAKE_CURRENT_SOURCE_DIR}
-                            -I ${CMAKE_CURRENT_BINARY_DIR}
-                            -o ${CMAKE_CURRENT_BINARY_DIR}/zlib1rc.obj
-                            -i ${CMAKE_CURRENT_SOURCE_DIR}/win32/zlib1.rc)
+                            --target=${ARCH_TARGET}
+                            -I ../zlib
+                            -I .
+                            -o zlib1rc.obj
+                            -i ../zlib/win32/zlib1.rc
+                            WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+			    )
     set(ZLIB_SRCS ${ZLIB_SRCS} ${CMAKE_CURRENT_BINARY_DIR}/zlib1rc.obj)
 endif(MINGW)
 
@@ -165,9 +174,9 @@
 
 if(NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL )
     install(TARGETS zlib
-        RUNTIME DESTINATION bin
-        ARCHIVE DESTINATION lib
-        LIBRARY DESTINATION lib )
+        RUNTIME DESTINATION bin/${CFS_ARCH_STR}
+        ARCHIVE DESTINATION ${LIB_SUFFIX}/${CFS_ARCH_STR}
+        LIBRARY DESTINATION ${LIB_SUFFIX}/${CFS_ARCH_STR} )
 endif()
 if(NOT SKIP_INSTALL_HEADERS AND NOT SKIP_INSTALL_ALL )
     install(FILES ${ZLIB_PUBLIC_HDRS} DESTINATION include)
