--- Makefile.install	2014-07-17 09:15:57.000000000 +0200
+++ /home/fwein/tmp/xianyi-OpenBLAS-f20c0f9/Makefile.install	2014-08-04 19:18:19.079129678 +0200
@@ -4,9 +4,10 @@
 include ./Makefile.system
 
 PREFIX ?= /opt/OpenBLAS
+CFS_LIB_TARGET ?= $(PREFIX)/lib
 
 OPENBLAS_INCLUDE_DIR := $(PREFIX)/include
-OPENBLAS_LIBRARY_DIR := $(PREFIX)/lib
+OPENBLAS_LIBRARY_DIR := $(CFS_LIB_TARGET)
 OPENBLAS_BINARY_DIR := $(PREFIX)/bin
 OPENBLAS_BUILD_DIR := $(CURDIR)
 
