--- CMakeLists.txt.org	2025-02-05 13:33:46
+++ CMakeLists.txt	2025-02-05 13:34:23
@@ -22,7 +22,7 @@
 endif()
 
 # Find hdf5
-find_package(HDF5 REQUIRED COMPONENTS C HL)
+# find_package(HDF5 REQUIRED COMPONENTS C HL)
 if(HDF5_FOUND)
     message(STATUS "Found HDF5: ${HDF5_INCLUDE_DIRS}")
     include_directories(${HDF5_INCLUDE_DIRS})
@@ -94,5 +94,5 @@
 # built.
 link_directories (${PROJECT_BINARY_DIR}/source  ${ZLIB_LIBRARIES} ${HDF5_LIBRARIES} ${HDF5_HL_LIBRARIES})
 
-add_subdirectory( examples )
-add_subdirectory( examples/MultiplePartition )
+#add_subdirectory( examples )
+#add_subdirectory( examples/MultiplePartition )
