--- source/gidpost.cpp.orig	2012-04-17 13:51:52.000000000 +0200
+++ source/gidpost.cpp	2012-04-17 13:53:23.000000000 +0200
@@ -12,6 +12,8 @@
 #include <math.h>
 #include <string.h>
 #include <stdlib.h>
+#include <iostream>
+#include <sstream>
 
 #include "gidpostInt.h"
 #include "gidpost.h"
@@ -118,10 +120,14 @@
       if (s == _ResultTypeInfo[i].dims[i])
         break;
     }
-    if (i == NMAX_DIMS)
-      printf("Invalid dimension %u for type %s\n", s, buffer);
-    else
-      sprintf(ptr, ":%u", s);
+    if (i == NMAX_DIMS) {
+      std::cerr << "Invalid dimension " << s << " for type " << buffer << std::endl;
+    }
+    else {
+      std::stringstream sstr;
+      sstr << s;
+      sprintf(ptr, ":%s", sstr.str().c_str());
+    }
   }
   return buffer;
 }
