--- sparspak/revrse.c.orig	2009-02-16 22:59:56.383796666 +0100
+++ sparspak/revrse.c	2009-02-16 23:00:09.223844974 +0100
@@ -6,7 +6,7 @@
 /* 'gen1wd' but it is not included in the source code.                       */
 /* in C, 'v' has subscripts 0,1,2,...,*nv-1.                                 */
 /*****************************************************************************/
-void revrse(integer *nv, integer *v)
+void revrse_(integer *nv, integer *v)
 {
    integer   dummy, 
          i=*nv/2, k;
