--- include/ilupack.h	2008-05-31 19:16:12.000000000 +0200
+++ include/ilupack.h	2008-08-15 18:29:32.000000000 +0200
@@ -4784,11 +4784,12 @@
 
 
 /* global variable to measure the timings of the components within ILUPACK */
+/*
 #define ILUPACK_secnds_length   10
     double ILUPACK_secnds[ILUPACK_secnds_length];
 #define ILUPACK_mem_length      12
     size_t ILUPACK_mem[ILUPACK_mem_length];
-
+*/
 
 #endif /* _ILU_PACK_H */
 
