--- src/cpp/flann/defines.h.orig	2014-02-19 20:34:28.465025014 +0100
+++ src/cpp/flann/defines.h	2014-02-19 20:34:54.320025010 +0100
@@ -89,14 +89,14 @@
     FLANN_INDEX_KDTREE_CUDA 	= 7,
 #endif
     FLANN_INDEX_SAVED 			= 254,
-    FLANN_INDEX_AUTOTUNED 		= 255,
+    FLANN_INDEX_AUTOTUNED 		= 255
 };
 
 enum flann_centers_init_t
 {
     FLANN_CENTERS_RANDOM = 0,
     FLANN_CENTERS_GONZALES = 1,
-    FLANN_CENTERS_KMEANSPP = 2,
+    FLANN_CENTERS_KMEANSPP = 2
 };
 
 enum flann_log_level_t
@@ -124,7 +124,7 @@
     FLANN_DIST_HAMMING         		= 9,
     FLANN_DIST_HAMMING_LUT			= 10,
     FLANN_DIST_HAMMING_POPCNT   	= 11,
-    FLANN_DIST_L2_SIMPLE	   		= 12,
+    FLANN_DIST_L2_SIMPLE	   		= 12
 };
 
 enum flann_datatype_t
@@ -144,7 +144,7 @@
 
 enum flann_checks_t {
     FLANN_CHECKS_UNLIMITED = -1,
-    FLANN_CHECKS_AUTOTUNED = -2,
+    FLANN_CHECKS_AUTOTUNED = -2
 };
 
 #ifdef __cplusplus
