--- CMakeLists.txt.org	2025-01-27 16:58:41
+++ CMakeLists.txt	2025-01-27 17:01:04
@@ -146,9 +146,12 @@
     endif(CUDA_FOUND)
 endif(BUILD_CUDA_LIB)
 
-find_package(PkgConfig REQUIRED)
-pkg_check_modules(LZ4 REQUIRED liblz4)
-include_directories(${LZ4_INCLUDE_DIRS})
+#find_package(PkgConfig REQUIRED)
+#pkg_check_modules(LZ4 REQUIRED liblz4)
+#include_directories(${LZ4_INCLUDE_DIRS})
+include_directories(${lz4_DIR}/include)
+link_directories(${lz4_DIR}/lib)
+link_libraries(lz4)
 
 #set the C/C++ include path to the "include" directory
 include_directories(BEFORE ${PROJECT_SOURCE_DIR}/src/cpp)
