--- src/cpp/CMakeLists.txt.org	2025-01-27 17:22:48
+++ src/cpp/CMakeLists.txt	2025-01-27 17:25:50
@@ -29,29 +29,29 @@
     set_property(TARGET flann_cuda_s PROPERTY COMPILE_DEFINITIONS FLANN_STATIC)
 endif()
 
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_COMPILER_IS_GNUCC)
-    add_library(flann_cpp SHARED dummy.c)
-    set_target_properties(flann_cpp PROPERTIES LINKER_LANGUAGE CXX)
-    target_link_libraries(flann_cpp -Wl,-whole-archive flann_cpp_s -Wl,-no-whole-archive)
+#if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_COMPILER_IS_GNUCC)
+#    add_library(flann_cpp SHARED dummy.c)
+#    set_target_properties(flann_cpp PROPERTIES LINKER_LANGUAGE CXX)
+#    target_link_libraries(flann_cpp -Wl,-whole-archive flann_cpp_s -Wl,-no-whole-archive)
 
-    if (BUILD_CUDA_LIB)
-	cuda_add_library(flann_cuda SHARED dummy.c)
-        set_target_properties(flann_cuda PROPERTIES LINKER_LANGUAGE CXX)
-        target_link_libraries(flann_cuda -Wl,-whole-archive flann_cuda_s -Wl,-no-whole-archive)
-        set_property(TARGET flann_cpp_s PROPERTY COMPILE_DEFINITIONS FLANN_USE_CUDA)
-    endif()
-else()
-    add_library(flann_cpp SHARED ${CPP_SOURCES})
-    target_link_libraries(flann_cpp ${LZ4_LINK_LIBRARIES})
-    # export lz4 headers, so that MSVC to creates flann_cpp.lib
-    set_target_properties(flann_cpp PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS YES)
-    if (BUILD_CUDA_LIB)
-	cuda_add_library(flann_cuda SHARED ${CU_SOURCES})
-        set_property(TARGET flann_cpp PROPERTY COMPILE_DEFINITIONS FLANN_USE_CUDA)
-    endif()
-endif()
+#    if (BUILD_CUDA_LIB)
+#	cuda_add_library(flann_cuda SHARED dummy.c)
+#        set_target_properties(flann_cuda PROPERTIES LINKER_LANGUAGE CXX)
+#        target_link_libraries(flann_cuda -Wl,-whole-archive flann_cuda_s -Wl,-no-whole-archive)
+#        set_property(TARGET flann_cpp_s PROPERTY COMPILE_DEFINITIONS FLANN_USE_CUDA)
+#    endif()
+#else()
+#    add_library(flann_cpp SHARED ${CPP_SOURCES})
+#    target_link_libraries(flann_cpp ${LZ4_LINK_LIBRARIES})
+#    # export lz4 headers, so that MSVC to creates flann_cpp.lib
+#    set_target_properties(flann_cpp PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS YES)
+#    if (BUILD_CUDA_LIB)
+#	cuda_add_library(flann_cuda SHARED ${CU_SOURCES})
+#        set_property(TARGET flann_cpp PROPERTY COMPILE_DEFINITIONS FLANN_USE_CUDA)
+#    endif()
+#endif()
 
-set_target_properties(flann_cpp PROPERTIES
+set_target_properties(flann_cpp_s PROPERTIES
    VERSION ${FLANN_VERSION}
    SOVERSION ${FLANN_SOVERSION}
    DEFINE_SYMBOL FLANN_EXPORTS
@@ -117,7 +117,7 @@
 
 
 install (
-    TARGETS flann_cpp flann_cpp_s
+    TARGETS flann_cpp_s
     EXPORT ${targets_export_name}
     INCLUDES DESTINATION include
     RUNTIME DESTINATION bin
@@ -138,7 +138,7 @@
 
 if (BUILD_C_BINDINGS)
     install (
-        TARGETS flann flann_s
+        TARGETS flann_s
         EXPORT ${targets_export_name}
         INCLUDES DESTINATION include
         RUNTIME DESTINATION bin
