--- src/cpp/flann/util/serialization.h.orig	2014-02-06 19:33:29.190196671 +0100
+++ src/cpp/flann/util/serialization.h	2014-02-06 19:36:57.637196639 +0100
@@ -89,8 +89,12 @@
 BASIC_TYPE_SERIALIZER(float);
 BASIC_TYPE_SERIALIZER(double);
 BASIC_TYPE_SERIALIZER(bool);
-
-
+#ifdef _MSC_VER
+BASIC_TYPE_SERIALIZER(unsigned __int64);
+#endif
+#ifdef __MINGW32__
+BASIC_TYPE_SERIALIZER(long long unsigned int);
+#endif
 
 // serializer for std::vector
 template<typename T>
