--- boost/lexical_cast.hpp.orig	2011-11-25 02:57:12.576737558 +0100
+++ boost/lexical_cast.hpp	2011-11-25 02:57:23.470039331 +0100
@@ -1181,7 +1181,7 @@
             bool shl_double(double val,T* out)
             {   using namespace std;
                 if (put_inf_nan(start,finish,val)) return true;
-                finish = start + sprintf(out,"%.*lg", static_cast<int>(boost::detail::lcast_get_precision<double >()), val );
+                finish = start + sprintf(out,"%.*g", static_cast<int>(boost::detail::lcast_get_precision<double >()), val );
                 return finish > start;
             }
 #ifndef __MINGW32__
@@ -1189,7 +1189,8 @@
             bool shl_long_double(long double val,T* out)
             {   using namespace std;
                 if (put_inf_nan(start,finish,val)) return true;
-                finish = start + sprintf(out,"%.*Lg", static_cast<int>(boost::detail::lcast_get_precision<long double >()), val );
+                finish = start + sprintf(out,"%.*g", static_cast<int>(boost::detail::lcast_get_precision<long double >()), val );
+
                 return finish > start;
             }
 #endif
