--- boost/multi_array/base.hpp.orig	2013-02-13 07:29:48.181569002 +0100
+++ boost/multi_array/base.hpp	2013-02-13 07:31:01.313146001 +0100
@@ -497,18 +497,22 @@
                      (start == index_bases[n] && extents[n] == 0)));
 
 #ifndef BOOST_DISABLE_ASSERTS
+#ifndef NDEBUG
       // finish marks the open side of the range, so it can go one past
       // the "far side" of the range (the top if stride is positive, the bottom
       // if stride is negative).
       index bound_adjustment = stride < 0 ? 1 : 0;
       BOOST_ASSERT(((index_bases[n] - bound_adjustment) <= finish) &&
         (finish <= (index_bases[n] + index(extents[n]) - bound_adjustment)));
+#endif
 #endif // BOOST_DISABLE_ASSERTS
 
 
       // the array data pointer is modified to account for non-zero
       // bases during slicing (see [Garcia] for the math involved)
       offset += start * strides[n];
+      new_strides[dim] = 0;
+      new_extents[dim] = 0;
 
       if (!current_range.is_degenerate()) {
 
