--- boost/math/tools/config.hpp.orig	2010-03-10 14:04:01.000000000 +0100
+++ boost/math/tools/config.hpp	2010-03-10 14:18:11.000000000 +0100
@@ -6,6 +6,11 @@
 #ifndef BOOST_MATH_TOOLS_CONFIG_HPP
 #define BOOST_MATH_TOOLS_CONFIG_HPP
 
+#ifdef __INTEL_COMPILER
+#include <mathimf.h>
+#define __builtin_fpclassify fpclassify
+#endif
+
 #include <boost/cstdint.hpp> // for boost::uintmax_t
 #include <boost/config.hpp>
 #include <boost/detail/workaround.hpp>
--- boost/functional/hash/detail/hash_float.hpp.orig	2010-02-23 15:58:11.000000000 +0100
+++ boost/functional/hash/detail/hash_float.hpp	2010-03-15 15:45:22.000000000 +0100
@@ -59,6 +59,11 @@
 
 #endif
 
+#ifdef __INTEL_COMPILER
+#include <mathimf.h>
+#endif
+
+
 namespace boost
 {
     namespace hash_detail
