--- boost/math/policies/error_handling.hpp.orig	2015-06-21 15:16:10.362983989 -0400
+++ boost/math/policies/error_handling.hpp	2015-06-21 15:56:34.295410881 -0400
@@ -677,7 +677,8 @@
 inline bool check_overflow(std::complex<T> val, R* result, const char* function, const Policy& pol)
 {
    typedef typename R::value_type r_type;
-   r_type re, im;
+   r_type re = r_type();
+   r_type im = r_type();
    bool r = check_overflow<r_type>(val.real(), &re, function, pol);
    r = check_overflow<r_type>(val.imag(), &im, function, pol) || r;
    *result = R(re, im);
