--- boost/unordered/detail/table.hpp.orig	2012-02-28 21:41:05.545081217 +0100
+++ boost/unordered/detail/table.hpp	2012-02-28 21:41:44.273061029 +0100
@@ -174,13 +174,13 @@
         }
 
         iterator& operator++() {
-            node_ = node_ = static_cast<node_pointer>(node_->next_);
+            node_ =  static_cast<node_pointer>(node_->next_);
             return *this;
         }
 
         iterator operator++(int) {
             iterator tmp(node_);
-            node_ = node_ = static_cast<node_pointer>(node_->next_);
+            node_ = static_cast<node_pointer>(node_->next_);
             return tmp;
         }
 
@@ -242,7 +242,7 @@
 
         c_iterator operator++(int) {
             c_iterator tmp(node_);
-            node_ = node_ = static_cast<node_pointer>(node_->next_);
+            node_ = static_cast<node_pointer>(node_->next_);
             return tmp;
         }
 
