--- boost/lexical_cast.hpp	2015-03-10 09:46:01.000000000 +0100
+++ boost/lexical_cast.hpp	2015-05-31 22:04:46.393498944 +0200
@@ -36,7 +36,7 @@
     template <typename Target, typename Source>
     inline Target lexical_cast(const Source &arg)
     {
-        Target result;
+        Target result = Target();
 
         if (!boost::conversion::detail::try_lexical_convert(arg, result)) {
             boost::conversion::detail::throw_bad_cast<Source, Target>();
